NOTES.UNIX

```diff
@@ -5,38 +5,72 @@
  For Unix/POSIX runtime systems on Windows, please see NOTES.WIN.
 
 
- Shared libraries and installation in non-standard locations
- -----------------------------------------------------------
-
- Binaries on Unix variants expect to find shared libraries in standard
- locations, such as /usr/lib, /usr/local/lib and some other locations
- configured in the system (for example /etc/ld.so.conf on some systems).
- If the libraries are installed in non-standard locations, binaries
- will not find them and therefore fail to run unless they get a bit of
- help from a defined RPATH or RUNPATH.  This can be applied by adding
- the appropriate linker flags to the configuration command, such as
- this (/usr/local/ssl was the default location for OpenSSL installation
- in versions before 1.1.0):
+ Shared libraries and installation in non-default locations
+ ----------------------------------------------------------
 
-    $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,/usr/local/ssl/lib
+ Binaries on some Unix variants expect to find shared libraries in
+ default locations, such as /lib, /usr/lib or possibly /usr/local/lib.
+ (what the default locations are depends on the system the libraries are
+ going to be installed in)
+ If the libraries are installed in non-default locations, binaries will
+ not find them and therefore fail to run unless they get a bit of help
+ from a defined runtime shared library search path.
 
- Because the actual library location may vary further (for example on
- multilib installations), there is a convenience variable in Makefile
- that holds the exact installation directory and that can be used like
- this:
+ OpenSSL's configuration scripts do NOT generally set the runtime shared
+ library search path for you.  It's therefore advisable to set it
+ explicitelly when configuring unless the libraries are to be installed
+ in directories that are known to be default.
 
-    $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,'$(LIBRPATH)'
+ Runtime shared library search paths are specified with different
+ linking options depending on operating system and versions thereof, and
+ are talked about differently in their respective documentation, where
+ variations of RPATH are the most usual (to be noted: ELF systems have
+ two such tags, more on that below).
+
+ Possible options to set the runtime shared library search path are:
 
- On modern systems using GNU ld.so, a better choice may be to use the
- new dtags, like this:
+    -rpath /whatever/path
+    -R /whatever/path
+    -Wl,-rpath,/whatever/path
```

Comment:
It's probably appropriate to add a comment that refers to OS. I mean e.g.
```
   -rpath /whatever/path    # Tue64
```
And why not list every one we know, i.e. add even HP-UX from below. For reference, on AIX there two options, -Wl,-bsvr4,-R,/whatever/path and -Wl,-libpath,/whatever/path:/usr/lib:/lib. However as additional *config* option -Wl,-R,/whatever/path, because -Wl,-bsvr4 is already passed.

Reply:
Just in case, as for -rpath being accounted to Tru64. I mean how -Wl,-rpath,/x is different from -rpath /x. -rpath works as compiler option on Tru64, e.g. cc -rpath ..., but not others. Others **have to** use -Wl.

Reply:
Does *only* Tru64 allow `cc` to take `-rpath`?

Reply:
Ah! IRIX compilers accept it too. And I fail to see that anybody else does. -R is accepted by Solaris compilers. BTW, plural in "compilers" means both vendor and gcc. So that gcc for Linux won't accept "naked" -R or -rpath as valid option, while gcc for Tru64 does accept -rpath and gcc for Solaris -R. Obviously in order to make it more interchangeable with vendor compiler.